packages/widget/src/utils/token.ts

```diff
@@ -7,3 +7,21 @@ export function tokenBalanceToNumber(
 ): number {
   return Number.parseFloat(utils.formatUnits(amount, decimals));
 }
+
+export function truncateBigNumberToString(
+  value: BigNumber,
+  decimals: number,
+  displayDecimals: number = 4
+): string {
+  const valueStr = utils.formatUnits(value, decimals);
+
+  const parts = valueStr.split('.');
```

Comment:
can we do here something like `Number(valueStr).toFixed(displayDecimals)`? You avoid having to split and slice stuff here. If `displayDecimals` is 0, we can just return the number

Reply:
@wainola This was an initial issue toFixed will round the balance resulting in having more balance than the user has. [Reference about toFixed](https://medium.com/@stheodorejohn/javascripts-tofixed-precision-in-your-hands-89ff981f50a9#:~:text=Rounding%20Behavior%3A%0AtoFixed()%20uses%20standard%20rounding%20behavior%2C%20so%20values%20are%20rounded%20to%20the%20nearest%20decimal%20place.%20Be%20aware%20that%20this%20may%20not%20always%20match%20your%20specific%20rounding%20requirements.)

Reply:
oh I see, thanks for the clarification